### PR TITLE
`ForIdentity` column of Relationship Templates and Tokens are not anonymized during identity deletion

### PR DESCRIPTION
**Pull request by MH321Productions**
# Readiness checklist

-   [x] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: Modules/Relationships/src/Relationships.Application/RelationshipTemplates/Commands/AnonymizeRelationshipTemplatesForIdentity/Handler.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        _relationshipTemplatesRepository = relationshipTemplatesRepository;
+        _applicationOptions = options.Value;
+    }
+
+    public async Task Handle(AnonymizeRelationshipTemplatesForIdentityCommand request, CancellationToken cancellationToken)
+    {
+        var relationshipTemplates = (await _relationshipTemplatesRepository.FindTemplates(RelationshipTemplate.IsFor(IdentityAddress.Parse(request.IdentityAddress)), cancellationToken)).ToList();
+
+        foreach (var relationshipTemplate in relationshipTemplates) relationshipTemplate.AnonymizeForIdentity(_applicationOptions.DidDomainName);
```
Comment (tnotheis) [inline Modules/Relationships/src/Relationships.Application/RelationshipTemplates/Commands/AnonymizeRelationshipTemplatesForIdentity/Handler.cs:24]: It's really hard to read if the main purpose of the loop is not at the beginning of the line.
```suggestion
        foreach (var relationshipTemplate in relationshipTemplates) 
            relationshipTemplate.AnonymizeForIdentity(_applicationOptions.DidDomainName);
```

File: Modules/Relationships/src/Relationships.Domain/Aggregates/RelationshipTemplates/RelationshipTemplate.cs [hunk 2]
```diff
@@ -69,6 +69,15 @@ public void AllocateFor(IdentityAddress identity, DeviceId device)
         Allocations.Add(new RelationshipTemplateAllocation(Id, identity, device));
     }
 
+    public void AnonymizeForIdentity(string didDomainName)
+    {
+        EnsureIsPersonalized();
```
Comment (tnotheis) [inline Modules/Relationships/src/Relationships.Domain/Aggregates/RelationshipTemplates/RelationshipTemplate.cs:74]: I don't think that an un-personalized template has to throw here. We should just ignore it.

File: Modules/Relationships/src/Relationships.Domain/DomainErrors.cs [hunk 3]
```diff
@@ -82,4 +82,10 @@ public static DomainError RelationshipTemplateNotAllocated()
         return new DomainError("error.platform.validation.relationship.relationshipTemplateNotAllocated",
             "The relationship template has to be allocated before it can be used to establish a relationship. Send a GET request to the /RelationshipTemplates/{id} endpoint to allocate the template.");
     }
+
+    public static DomainError RelationshipTemplateNotPersonalized()
```
Comment (tnotheis) [inline Modules/Relationships/src/Relationships.Domain/DomainErrors.cs:86]: Remember to remove this after removing the code that throws the error.

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Relationships/src/Relationships.Domain/Aggregates/RelationshipTemplates/RelationshipTemplate.cs (hunk 2 above)
Comment (MH321Productions) [reply Modules/Relationships/src/Relationships.Domain/Aggregates/RelationshipTemplates/RelationshipTemplate.cs:74]: So it should just return silently?

**Review by tnotheis — APPROVED** [review]

**Pull request merged**